packages/core/content-type-builder/admin/src/utils/timeFormat.ts

```diff
@@ -0,0 +1,55 @@
+type TimeChangeHandler = (params: {
+  target: { name: string; value: string | undefined; type: string };
+}) => void;
+
+type TimeChangeParams = {
+  value: string | undefined;
+  onChange: TimeChangeHandler;
+  name: string;
+  type: string;
+};
+
+// The backend send a value which has the following format: '00:45:00.000'
+// or the time picker only supports hours & minutes so we need to mutate the value
+const removeSeconds = (time: string): string => {
+  const [hours, minutes] = time.split(':');
+  return `${hours}:${minutes}`;
+};
+
+// we need to send back the value with the same '00:45:00.000' format
+const addSecondsAndMilliseconds = (time: string): string => {
+  return time.split(':').length === 2 ? `${time}:00.000` : time;
+};
+
+const formatTimeForInput = (value: string | undefined): string | undefined => {
+  if (!value) return undefined;
```

Comment:
you can choose to do this  you don't have to explicitly return undefined
```suggestion
  if (!value) return;
```
or just use 

```
if (value) {
  return value.split(':').length > 2 ? removeSeconds(value) : value;
}
```